pkg/controllers/controllers.go

```diff
@@ -87,5 +93,16 @@ func NewControllers(
 		nodeclaimdisruption.NewController(clock, kubeClient, cloudProvider),
 		status.NewController[*v1.NodeClaim](kubeClient, mgr.GetEventRecorderFor("karpenter")),
 		status.NewController[*v1.NodePool](kubeClient, mgr.GetEventRecorderFor("karpenter")),
+		// status.NewGenericObjectController[*corev1.Node](kubeClient, mgr.GetEventRecorderFor("karpenter")),
+		// status.NewGenericObjectController[*corev1.Pod](kubeClient, mgr.GetEventRecorderFor("karpenter")),
+	}
+
+	// The cloud proivder must define status condation for the node repair controller to used for dectecting unhealthy nodes
+	if len(cloudProvider.RepairPolicy()) != 0 && !options.FromContext(ctx).FeatureGates.NodeRepair {
+		controllers = append(controllers, health.NewController(kubeClient, cloudProvider, clock))
+	} else {
+		log.FromContext(ctx).V(1).Info("node repair has been disabled")
```

Comment:
I don't think you want to log that it's disabled -- you probably want to log the opposite since this is disabled by default.

Also, you generally want to log when you are doing actions, not when you aren't doing them

Reply:
I was trying to follow how we approached the drift feature flag. In there, we would log each time we removed a status condition, and I was thinking it made sense to log the feature setting once on startup. I removed it now